packages/graphql-language-service-server/src/startServer.js

```diff
@@ -17,67 +17,88 @@ import {
 } from './MessageProcessor';
 
 import {
+  IPCMessageReader,
   IPCMessageWriter,
   SocketMessageReader,
   SocketMessageWriter,
 } from 'vscode-jsonrpc';
 
-type ServerOptions = {
+type Options = {
   port?: number,
+  method?: string,
+  configDir?: string,
 };
 
-export default (async function startServer(
-  configDir: ?string,
-  options?: ServerOptions,
-): Promise<void> {
-  if (options && options.port) {
-    // Socket protocol support
-    const socket = net.createServer(client => {
-      client.setEncoding('utf8');
-      const messageReader = new SocketMessageReader(client);
-      const messageWriter = new SocketMessageWriter(client);
+export default (async function startServer(options?: Options): Promise<void> {
+  if (!options || !options.configDir) {
+    process.stderr.write('--configDir is required!');
+    process.exit(1);
+  }
+
+  const configDir = options.configDir;
+  if (options && options.method) {
+    switch (options.method) {
+      case 'node':
+        connectWithNodeIPC(configDir);
+        break;
+      case 'stream':
+        connectWithStream(configDir);
+        break;
+      case 'socket':
+        connectWithSocket(configDir, options.port);
+        break;
+    }
+  }
+});
+
+function connectWithSocket(configDir: string, port: number): Promise<void> {
+  // Socket protocol support
+  const socket = net.createServer(client => {
+    client.setEncoding('utf8');
+    const messageReader = new SocketMessageReader(client);
+    const messageWriter = new SocketMessageWriter(client);
 
-      client.on('end', () => {
-        socket.close();
-        process.exit(0);
-      });
+    client.on('end', () => {
+      socket.close();
+      process.exit(0);
+    });
 
-      messageReader.listen(message => {
-        try {
-          if (message.id != null) {
-            processIPCRequestMessage(message, configDir, messageWriter);
-          } else {
-            processIPCNotificationMessage(message, messageWriter);
-          }
-        } catch (error) {
-          // Swallow error silently.
+    messageReader.listen(message => {
+      try {
+        if (message.id != null) {
+          processIPCRequestMessage(message, configDir, messageWriter);
+        } else {
+          processIPCNotificationMessage(message, messageWriter);
         }
-      });
+      } catch (error) {
+        // Swallow error silently.
```

Comment:
(I know it was like this before, but it might be interesting to do something with the error — log it?)